### ISSUE TITLE
Add Station Number in Shanghai Metro

### ISSUE THREAD

**Issue by XHG78999** [wongchito/RailMapGenerator#220]
Shentong Metro Co.Ltd has test many new features in Line 10 (Only phase 1), like Station Number.
The Station Number is in Shanghai Metro Line 10 short-lived (Not use in after lines), but now it is exist in real world.

### About details of Station Number
As the real image, you can see the details.
![image](https://user-images.githubusercontent.com/53939187/149770475-a3a8ec13-a781-4216-a590-07a118b5e789.png)
![image](https://user-images.githubusercontent.com/53939187/149770138-ed0e73c8-8b8c-4a23-9fee-c05ea28a1133.png)
The station number is only show at the station name map.
Style is in the image.

**Comment by thekingofcity** [comment]
Yes, this is a long-unseen feature. I postpone this as it requires some work and it didn't manage to survive later. Happy to see someone bring it out and I will work on this.

The only problem will be the `info_panel_type` where current `sh` and `sh2020` mean railmap type instead of subtle variants. What will be the best solution to have a line 10 variant? Another `info_panel_type`?

Also, would you like to take this job? It will be pretty straightforward since you can take `gzmtr`'s canvas and panel as the reference.

**Comment by thekingofcity** [comment]
> About the style between shmetro and sh2020, I think it is only can used in shmetro old, because in Line 10 phase 2 there are no station number.

No station numbers even on `runin` canvas?

**Comment by XHG78999** [comment]
> 

Yes.

**Comment by thekingofcity** [comment]
That's not the case for stations in Yangpu, would you like to upload some photos showing the latest `runin`?

**Comment by XHG78999** [comment]
> That's not the case for stations in Yangpu, would you like to upload some photos showing the latest `runin`?

Right, I think in many years Shentong will remove them. Now the Xinjiangwancheng is replaced to no station number. Or like the Exit 4 of Longbai Xincun, thats the light sign is something wrong and then it be replaced a new sign have no station number.

But when I search the images, I see:
![7145300520487285350](https://user-images.githubusercontent.com/53939187/150456921-9cc8dab0-e24d-49ae-a5f3-07ecd226d3cc.jpg)
(Sorry for bad quality.) This photo is taked on 2011. 
Look at the `runin`! It **have** station numbers! 
It means that long long ago there are station numbers in `runin`.

**Comment by XHG78999** [comment]
I would like to do that and I am doing.

**Comment by XHG78999** [comment]
That's a plan:

1. At first, we will add station number to the staton name map. It will do on a branch or a local version.
2. Next, we will add station number to the runin map.
3. Fix bugs and change some digit.

I will try my best to learn React.
Thanks. I think you can add `feature` tag to it.

**Comment by XHG78999** [comment]
That's a plan:

1. At first, we will add station number to the staton name map. It will do on a branch or a local version.
2. Next, we will add station number to the runin map.
3. Fix bugs and change some digit.

I will try my best to learn React.
Thanks. I think you can add `feature` tag to it.

**Comment by thekingofcity** [comment]
> Right, I think in many years Shentong will remove them. Now the Xinjiangwancheng is replaced to no station number. Or like the Exit 4 of Longbai Xincun, thats the light sign is something wrong and then it be replaced a new sign have no station number.

So how about a `sh_10` in `info_panel_type` which is based on `sh` and has station number in `runin` and `railmap`? (Not quite sure what it will be like in `indoor`)

> I would like to do that and I am doing.

Great to see a new contributor! You're welcome to open a pull request (even before finishing to ask for code help) and I'll help you to publish it to the [uat](https://github.com/wongchito/uat-rail-map-generator)

> I think you can add a feature tag to it.

I do not follow. Isn't it currently have the right tag?

**Comment by thekingofcity** [comment]
*A kind reminder*: Use existing UI components as much as possible as there is a major overhaul on UI in v5 :)

**Comment by XHG78999** [comment]
> _A kind reminder_: Use existing UI components as much as possible because there is a major overhaul on UI in v5 :)

I know. I'll remind you: Please disable this in the round line.

**Comment by XHG78999** [comment]
@thekingofcity I would like to discuss together to know some in the code. My Email is me@xhg78999.tk .

**Comment by thekingofcity** [comment]
Do you already contact me with IM?

**Comment by XHG78999** [comment]
Sorry I am busy now. My computer is something(system) broken.... I am try my best to rescue it......-------- 原始邮件 --------发件人： thekingofcity ***@***.***>日期： 2022年1月23日周日 晚上8:14收件人： wongchito/RailMapGenerator ***@***.***>抄送： XHG78999 ***@***.***>, Author ***@***.***>主    题： Re: [wongchito/RailMapGenerator] Add Station Number in Shanghai Metro (Issue #220)
Do you already contact me with IM?

**Comment by XHG78999** [comment]
> Do you already contact me with IM?
> 

We also can use Email or Slack? My V*N and SSR doesn't work there.

**Comment by thekingofcity** [comment]
How about Slack workspace?

https://join.slack.com/t/railmapgenerator/shared_invite/zt-12bep9j92-10xVOfyCjkfWQaNpC7oShw

**Comment by thekingofcity** [comment]
![Image](https://github.com/user-attachments/assets/977cbf66-3aca-442b-bc19-c4e0430d6db1)

Thanks (L2L3延安西路站)[https://space.bilibili.com/3546657385875973]
